src/communicator/communicator.py

```diff
@@ -50,34 +53,78 @@ def __init__(
         ),
     ):
         """Creates a new Communicator object"""
-        Thread.__init__(self)
         self._configuration = sumo_configuration
         self._port = sumo_port
         self._components = components if components is not None else []
         self._max_tick = max_tick
 
-    def run(self):
+    def run(self) -> str:
+        process = self._run.delay(
+            max_tick=self._max_tick,
+            components_pickle=pickle.dumps(self._components),
+            configuration=self._configuration,
+            port=self._port,
+        )
+        return process.id
+
+    @celery.task(bind=True, ignore_result=False)
+    def _run(
+        self: Task,
+        max_tick: int,
+        components_pickle: bytes,
+        configuration: str,
+        port: int,
+    ):
         """Starts sumo (no gui) and connects using traci. The connection has the `default` label."""
 
-        traci.start([checkBinary("sumo"), "-c", self._configuration], port=self._port)
+        components = pickle.loads(components_pickle)
```

Comment:
I am confused: Why do we need to pass the components serialized?

Reply:
We have to pass a serialized version of the components because of the functionality of celery. Celery runs the started processes in a new service/container, "celery-worker". To run the method call as a new process, celery passes the function arguments via an internet protocol. It only allows primitive datatypes, so we must serialize the components.